subsys/net/ip/net_context.c

```diff
@@ -690,13 +709,16 @@ int net_context_connect(struct net_context *context,
 	NET_ASSERT(addr);
 	NET_ASSERT(PART_OF_ARRAY(contexts, context));
 
+	k_mutex_lock(&context->lock, K_FOREVER);
+
 	if (!net_context_is_used(context)) {
-		return -EBADF;
+		ret = -EBADF;
+		goto out;
 	}
 
 	ret = bind_default(context);
 	if (ret) {
-		return ret;
+		goto out;
```

Comment:
Ok, here a goto out is used, this is as good as the above. 'out' could be renamed 'unlock', as that is what it does (bikeshedding...).